### PR TITLE
Change the classes with shortcut

### PR DESCRIPTION
**Pull request by jannisCode**
This PR finishes the PR started by @Wittmaxi , who also did most of the work here: https://github.com/eclipse-platform/eclipse.platform.ui/pull/1874
The objective of @Wittmaxi ´s PR is to be able to expand the functionality of the nextTab shortcut (default crtl + pgdown) so that after reaching the last tab, one more click changes the active tab back to the first one. 

## What this PR does
When there are too many tabs open,  #1874 still went back to the first tab, ignoring the remaining tabs stored in the chevron (the >> container). 
image
With the changes from this PR, when there are stored tabs in the chevron, clicking once more after the last tab has the same functionality as the default eclipse, going into the chevron.

https://github.com/user-attachments/assets/21be2294-6841-4b78-b5bd-3b25b00794cb

### CODE REVIEW

File: bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java [hunk 1]
```diff
@@ -38,20 +40,90 @@ public class TraversePageHandler extends WidgetMethodHandler {
...
+		}
+		return currentFolder.equals(lastFolder);
+	}
+
+	/**
+	 * checks if there are any hidden items
+	 *
+	 * @param folder
+	 * @return if there are any items which are not shown in the folder (tabs in the
+	 *         >> column)
```
Comment (Wittmaxi) [inline bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java]: the ">> column" is called "the chevron"

File: bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java [hunk 2]
```diff
@@ -38,20 +40,90 @@ public class TraversePageHandler extends WidgetMethodHandler {
...
+		} else {
+			lastFolder = folder.getItem(0);
+		}
+		return currentFolder.equals(lastFolder);
+	}
+
+	/**
+	 * checks if there are any hidden items
+	 *
+	 * @param folder
```
Comment (Wittmaxi) [inline bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java]: please describe further

File: bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java [hunk 3]
```diff
@@ -38,20 +40,90 @@ public class TraversePageHandler extends WidgetMethodHandler {
...
+	 * @return if there are any items which are not shown in the folder (tabs in the
+	 *         >> column)
+	 */
+	private boolean areHiddenItems(CTabFolder folder) {
+		for (CTabItem i : folder.getItems()) {
```
Comment (Wittmaxi) [inline bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java]: Use https://www.geeksforgeeks.org/stream-anymatch-java-examples/

File: bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java [hunk 4]
```diff
@@ -38,20 +40,90 @@ public class TraversePageHandler extends WidgetMethodHandler {
...
 
+
+	private int getTraversalDirection(boolean direction) {
+		return direction ? SWT.TRAVERSE_PAGE_NEXT : SWT.TRAVERSE_PAGE_PREVIOUS;
+	}
+
+	/**
+	 * Sets the current selection to the second-to-last item in the given direction.
+	 *
+	 * @param folder
```
Comment (Wittmaxi) [inline bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java]: describe folder and forward

**Review by Wittmaxi — CHANGES_REQUESTED** [review]
Small nitpicks. I can check functionality later and merge if it works

File: bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java (hunk 1 above)
Comment (Wittmaxi) [reply bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java]: https://ux.stackexchange.com/questions/105135/difference-between-caret-and-chevron

Comment (jannisCode) [reply bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java]: done

File: bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java (hunk 2 above)
Comment (jannisCode) [reply bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java]: done

File: bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java (hunk 3 above)
Comment (jannisCode) [reply bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java]: done

File: bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java (hunk 4 above)
Comment (jannisCode) [reply bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java]: done

File: bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java [hunk 5]
```diff
@@ -28,30 +28,84 @@
...
-
 	@Override
 	public final Object execute(final ExecutionEvent event) {
 		Control focusControl = Display.getCurrent().getFocusControl();
 		if (focusControl != null) {
-			int traversal = "next".equals(methodName) ? SWT.TRAVERSE_PAGE_NEXT : SWT.TRAVERSE_PAGE_PREVIOUS; //$NON-NLS-1$
+			boolean forward = "next".equals(methodName); //$NON-NLS-1$
+			int traversal = getTraversalDirection(forward);
 			Control control = focusControl;
+
```
Comment (Wittmaxi) [inline bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java]: empty diff

File: bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java [hunk 6]
```diff
@@ -28,30 +28,84 @@
...
 			do {
+				if (control instanceof CTabFolder folder && isFinalItemInCTabFolder(folder, forward)
+						&& !areHiddenItems(folder)) {
+					loopToSecondToFirstItemInCTabFolder(folder, forward);
```
Comment (Wittmaxi) [inline bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java]: "loopToSecondToFirstItemInCTabFolder" is not a desirable name. Are you referring to the "second to last" item or to the "second" item?

File: bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java [hunk 7]
```diff
@@ -28,30 +28,84 @@
...
+	private boolean areHiddenItems(CTabFolder folder) {
+		CTabItem[] items = folder.getItems();
+		for (CTabItem i : items) {
+			if (!i.isShowing()) {
+				return true;
+			}
+		}
+
+		return false;
+	}
```
Comment (Wittmaxi) [inline bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java]: by convention, please add a line here

File: bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java [hunk 8]
```diff
@@ -28,30 +28,84 @@
...
 				control = control.getParent();
 			} while (control != null);
 		}
-
 		return null;
 	}
 
+	private boolean areHiddenItems(CTabFolder folder) {
+		CTabItem[] items = folder.getItems();
+		for (CTabItem i : items) {
```
Comment (Wittmaxi) [inline bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java]: use this https://stackoverflow.com/questions/31419931/how-do-you-check-if-a-list-contains-an-element-that-matches-some-predicate

File: bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java [hunk 9]
```diff
@@ -70,5 +124,4 @@ protected Method getMethodToExecute() {
 		}
 		return null;
 	}
-
-}
+}
```
Comment (Wittmaxi) [inline bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java]: please keep the diff clean

File: bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java [hunk 10]
```diff
@@ -28,30 +28,84 @@
...
+		if (forward) {
+			folder.showItem(folder.getItem(0));
+			folder.setSelection(1);
+		} else {
+			int itemCount = folder.getItemCount();
+			folder.setSelection(itemCount - 2);
+		}
+	}
+
+	/**https://github.com/jannisCode/eclipse.jdt.ui.git
```
Comment (Wittmaxi) [inline bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java]: By convention, the first line of javadoc comments is empty (atleast that's what I have seen everywhere else)

File: bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java [hunk 11]
```diff
@@ -28,30 +28,84 @@
...
+		}
+
+		return false;
+	}
+	private int getTraversalDirection(boolean direction) {
+		return direction ? SWT.TRAVERSE_PAGE_NEXT : SWT.TRAVERSE_PAGE_PREVIOUS;
+	}
+
+	/**
+	 * Sets the current selection to the second-to-last item in the given direction.
```
Comment (Wittmaxi) [inline bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java]: second to last? or second to first?

**Review by Wittmaxi — CHANGES_REQUESTED** [review]

File: bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java (hunk 5 above)
Comment (jannisCode) [reply bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java]: done

File: bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java (hunk 6 above)
Comment (jannisCode) [reply bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java]: done

File: bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java (hunk 8 above)
Comment (jannisCode) [reply bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java]: done

File: bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java (hunk 10 above)
Comment (jannisCode) [reply bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java]: done

File: bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java (hunk 11 above)
Comment (jannisCode) [reply bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/handlers/TraversePageHandler.java]: done

**Pull request closed without merging**